mountpoint-s3-fs/src/manifest/csv_reader.rs

```diff
@@ -0,0 +1,79 @@
+use super::{DbEntry, ManifestError};
+use csv::{DeserializeRecordsIntoIter, ReaderBuilder};
+use std::io::Read;
+
+pub struct CsvReader<R: Read> {
+    reader: DeserializeRecordsIntoIter<R, DbEntry>,
+}
+
+impl<R: Read> CsvReader<R> {
+    pub fn new(reader: R) -> Self {
+        let reader = ReaderBuilder::new()
+            .has_headers(false)
+            .from_reader(reader)
+            .into_deserialize();
+        Self { reader }
+    }
+}
+
+impl<R: Read> Iterator for CsvReader<R> {
+    type Item = Result<DbEntry, ManifestError>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.reader.next().map(|item| item.map_err(ManifestError::from))
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use test_case::test_case;
+
+    #[test_case("\"key1\",\"\"\"etag1\"\"\",1024\n\"key2\",\"\"\"etag2\"\"\",2048\n", &[
```

Comment:
Could we use raw string literals here? This is very hard to read

Reply:
Agreed. Maybe even constants with raw multiline literals.